dmrpp_generator/main.py

```diff
@@ -67,44 +66,65 @@ def process(self):
         collection_files = collection.get('files', [])
+        collection_meta = collection.get('meta', {})
+        dmrpp_meta = collection_meta.get('dmrpp', {})
         buckets = self.config.get('buckets')
         granules = self.input['granules']
         for granule in granules:
             dmrpp_files = []
             for file_ in granule['files']:
                 if not match(f"{self.processing_regex}$", file_['filename']):
                     continue
-                output_file_path = self.dmrpp_generate(file_['filename'])
-                output_file_basename = os.path.basename(output_file_path)
-                if output_file_path:
-                    dmrpp_file = {
+                output_file_paths = self.dmrpp_generate(input_file=file_['filename'],
+                                                       dmrpp_meta=dmrpp_meta)
+                for output_file_path in output_file_paths:
+                    output_file_basename = os.path.basename(output_file_path)
+                    if output_file_path:
+                        dmrpp_file = {
                         "name": os.path.basename(output_file_path),
                         "path": self.config.get('fileStagingDir'),
                         "url_path": file_.get('url_path', self.config.get('fileStagingDir')),
                         "bucket": self.get_bucket(output_file_basename, collection_files, buckets)['name'],
                         "size": os.path.getsize(output_file_path),
                         "type": self.get_file_type(output_file_basename, collection_files)
-                    }
-                    prefix = os.path.dirname(file_['filepath'])
-                    dmrpp_file['filepath'] = f'{prefix}/{dmrpp_file["name"]}'
-                    dmrpp_file['filename'] = f's3://{dmrpp_file["bucket"]}/{dmrpp_file["filepath"]}'
-                    dmrpp_files.append(dmrpp_file)
-                    self.upload_file_to_s3(output_file_path, dmrpp_file['filename'])
+                        }
+                        dmrpp_file['filepath'] = f"{dmrpp_file['url_path'].rstrip('/')}/{dmrpp_file['name']}".lstrip('/')
+                        dmrpp_file['filename'] = f's3://{dmrpp_file["bucket"]}/{dmrpp_file["filepath"]}'
+                        dmrpp_files.append(dmrpp_file)
+                        self.upload_file_to_s3(output_file_path, dmrpp_file['filename'])
             granule['files'] += dmrpp_files
         return self.input
 
-    def dmrpp_generate(self, input_file, local=False):
+    @staticmethod
+    def get_dmrpp_command(dmrpp_meta, input_path, output_filename):
+        """
+        Getting the command line to create DMRPP files
+        """
+        dmrpp_meta = dmrpp_meta if isinstance(dmrpp_meta, dict) else {}
+        create_missing_cf = dmrpp_meta.get('create_missing_cf') in ['true', '1'] or os.getenv('create_missing_cf'.upper()) in ['true', '1']
+        create_missing_cf_option = '-M -b' if create_missing_cf else '-b'
```

Comment:
I'd like this to be scalable, so that we can add more switches with minimal change. For example
`def get_command(self, config, path, file_name):
        switches = ""

        data = None
        if config != None: data = json.loads(config)

        for key in self.switch_map:
            add_switch = False
            if data:
                try:   
                    add_switch = data['config']['meta']['dmrpp'][key] == True
                except KeyError:
                    pass               
            else:
                add_switch = os.getenv(key.upper(), 'False').lower() in ['true', '1']
                           
            if add_switch: switches = switches.join(f" {self.switch_map[key]}")

        return f'get_dmrpp{switches} -b {path} -o {file_name}.dmrpp {os.path.basename(file_name)}'`

Would require us to only add 
`switch_map = {
        "create_missing_cf": "-M"
        "add_foo": ' -foo'
    }`
For another switch of name 'foo'